diff --git a/c/avalontoolkit-src/avalontoolkit/SourceDistribution/common/layout.c b/c/avalontoolkit-src/avalontoolkit/SourceDistribution/common/layout.c
index f08b14449..ae00f8d9a 100644
--- a/c/avalontoolkit-src/avalontoolkit/SourceDistribution/common/layout.c
+++ b/c/avalontoolkit-src/avalontoolkit/SourceDistribution/common/layout.c
@@ -68,6 +68,12 @@
 
 typedef double point[2];
 
+int FloodInvertColor(struct reaccs_molecule_t *mp,
+                     neighbourhood_t          *nbp,
+                     int start_index,
+                     int fragmentColor,
+                     struct reaccs_bond_t *flipBond);
+int ChangeAtomColors(struct reaccs_molecule_t *mp, int fromColor, int toColor);
 static void LogString(char *message)
 {
    FILE *fp;
diff --git a/c/avalontoolkit-src/avalontoolkit/SourceDistribution/common/shortcut.c b/c/avalontoolkit-src/avalontoolkit/SourceDistribution/common/shortcut.c
index 23b13dd43..f4cd4fd6b 100644
--- a/c/avalontoolkit-src/avalontoolkit/SourceDistribution/common/shortcut.c
+++ b/c/avalontoolkit-src/avalontoolkit/SourceDistribution/common/shortcut.c
@@ -40,13 +40,14 @@
 #include <stdlib.h>
 #include <stdio.h>
 #include <string.h>
+#include <ctype.h>
 
 #include "shortcut.h"
 #include "smi2mol.h"
 #include "utilities.h"
 #include "ssmatch.h"
 #include "stereo.h"
-
+#include "perceive.h"
 struct aa_class_string_t
 {
     char *class_string;         // identifier of this atom environment
@@ -242,7 +243,7 @@ static struct shortcut_input_t shortcut_input[] =
          "AMINO_C",	"",	        ""},
     };
 
-static int StripRAtoms(struct reaccs_molecule_t *mp, char *code)
+static void StripRAtoms(struct reaccs_molecule_t *mp, char *code)
 /*
  * Removes the R-atoms from the query molecule mp and color the attached atoms with their mass if any.
  * This trick is used to remember the attachment position.
