diff --git a/cpp/coordgen-src/coordgen/CoordgenConfig.hpp b/cpp/coordgen-src/coordgen/CoordgenConfig.hpp
index 5f1e66840..781cfb948 100644
--- a/cpp/coordgen-src/coordgen/CoordgenConfig.hpp
+++ b/cpp/coordgen-src/coordgen/CoordgenConfig.hpp
@@ -1,22 +1,7 @@
 #pragma once
 
-#ifndef STATIC_COORDGEN
+#include <coordgenlibs_export.hpp>
 
-#ifdef WIN32
-#ifdef IN_COORDGEN
-#define EXPORT_COORDGEN __declspec(dllexport)
-#else
-#define EXPORT_COORDGEN __declspec(dllimport)
-#endif // IN_COORDGEN
-
-#else
-
-#define EXPORT_COORDGEN __attribute__((visibility("default")))
-
-#endif // WIN32
-
-#else
-
-#define EXPORT_COORDGEN
-
-#endif // STATIC_COORDGEN
+#ifndef EXPORT_COORDGEN
+#define EXPORT_COORDGEN COORDGENLIBS_EXPORT
+#endif
