diff --git a/cpp/maeparser-src/maeparser/MaeParserConfig.hpp b/cpp/maeparser-src/maeparser/MaeParserConfig.hpp
index 570c743e9..cc18b471a 100644
--- a/cpp/maeparser-src/maeparser/MaeParserConfig.hpp
+++ b/cpp/maeparser-src/maeparser/MaeParserConfig.hpp
@@ -1,21 +1,7 @@
 #pragma once
 
-#ifndef STATIC_MAEPARSER
+#include <maeparser_export.hpp>
 
-#ifdef WIN32
-#ifdef IN_MAEPARSER
-#define EXPORT_MAEPARSER __declspec(dllexport)
-#else
-#define EXPORT_MAEPARSER __declspec(dllimport)
-#endif // IN_MAEPARSER
-
-#else
-
-#define EXPORT_MAEPARSER __attribute__((visibility("default")))
-#endif // WIN32
-
-#else
-
-#define EXPORT_MAEPARSER
-
-#endif // STATIC_MAEPARSER
+#ifndef EXPORT_MAEPARSER
+#define EXPORT_MAEPARSER MAEPARSER_EXPORT
+#endif
